### ISSUE TITLE
User session and access token not present on the request for ListSkillItemCreatedRequest

### ISSUE THREAD

**Issue by Manav-Seth** [stoiveyp/Alexa.NET.ListManagement#10]
Hello, I am trying to use Alexa Account linking to get access tokens sent to the alexa skill through shopping list updates, since I am trying to write a skill for my personal household use to sync Alexa shopping list to a list tracking app me and my wife use. I have already done something similar for Wunderlist (and it works) but Wunderlist is shutting down in May :( Hence I am trying out another app (this time ToDo) but ToDo can only work if I actually link account since the access token for ToDo is not valid forever (like its the case with Wunderlist). 
But looks like, ListSkillItemCreatedRequest intent request does not have SkillRequest.Session object and hence I am not able to get the access token (Session.User.AccessToken). I have verified that the access token is present for an IntentRequest. Is this functionality missing? What would it take to add it? I can also look at adding it, to contribute.

**Comment by stoiveyp** [comment]
Good evening

Thanks for reaching out. Just want to clarify the issue that you're having. It sounds like you're trying to get the access token from your skill request. for the ListManagementClient

The Alexa.NET request sent to your skill is a SkillRequest object - which is your top level object, and then SkillRequest.Request - a property that contains the information itself.

The token you're after is in the new SkillRequest.Context.System.ApiAccessToken property, and is part of every valid request sent to your skill. The SkillRequest.Session object only exists as part of an active conversation with the user, and wouldn't be set as part of a list management call. But you still have the api token available in the context system object.

I hope this clears things up a bit and helps you get the functionality you need.

Thanks again for reaching out

Steven
________________________________

**Comment by Manav-Seth** [comment]
Thank you so much for a lightning fast response, even during the current situation due to the global pandemic. Thanks!
I assumed that the access token should be present in all requests, regardless of the intent type (that is what the documentation on amazon says). Moreover, there are existing skills like ToDoist, Any.Do, etc which somehow are able to achieve this - getting notifications for items being added to alexa lists and also able to send the items to their platform - unless, they too have a never expiring access token (like I was able to achieve the same experience for my own personal skill with wunderlist). Any idea how to achieve this then?

**Comment by stoiveyp** [comment]
Sorry, I'm a little unsure of what you're asking me about now? We've said where the Alexa API access token will (context instead of session). If you want the list event to interact with another platform that would require third party account linking - something you have to set up in your skill first (and you'd have to get information from the platform to do it)

Would you mind clarifying?

**Comment by Manav-Seth** [comment]
I might not be clear - apologies.
I am only referring to account linking - I already am using the ApiAccessToken successfully. 

I have set up account linking and its working as expected. I have verified that for a normal intent request (when the request type is IntentRequest), I do get the access token (I am referring to input.Session.User.AccessToken). I am not getting this for the ListSkillItemCreatedRequest - since in case of ListSkillItemCreatedRequest, input.Session is null. Note: input is variable of type SkillRequest. 
Does that make sense now? Apologies for the back and forth.
